File: cases/taint/cpp/return-relay-two-hop-negative/return_relay_two_hop_negative.cpp
int dfb_source() { // DFB-SOURCE: return-two-hop-negative-input
    return 1;
}

int first_relay(int value) { // DFB-WITNESS: return-two-hop-negative-first
    return value;
}

int second_relay(int value) { // DFB-WITNESS: return-two-hop-negative-second
    return first_relay(value);
}

void dfb_sink(int value) {} // DFB-SINK: return-two-hop-negative-sink

void run() {
    int result = second_relay(dfb_source());
    dfb_sink(0);
}

File: cases/taint/rust/return-relay-two-hop-negative/return_relay_two_hop_negative.rs
fn dfb_source() -> i32 { // DFB-SOURCE: return-two-hop-negative-input
    1
}

fn first_relay(value: i32) -> i32 { // DFB-WITNESS: return-two-hop-negative-first
    value
}

fn second_relay(value: i32) -> i32 { // DFB-WITNESS: return-two-hop-negative-second
    first_relay(value)
}

fn dfb_sink(value: i32) {} // DFB-SINK: return-two-hop-negative-sink

fn run() {
    let result = second_relay(dfb_source());
    dfb_sink(0);
}
